platform/android/java_godot_lib_jni.cpp

```diff
@@ -540,6 +540,23 @@ JNIEXPORT void JNICALL Java_org_godotengine_godot_GodotLib_onNightModeChanged(JN
 	}
 }
 
+JNIEXPORT void JNICALL Java_org_godotengine_godot_GodotLib_filePickerCallback(JNIEnv *env, jclass clazz, jboolean p_ok, jobjectArray p_selected_paths, jint p_filter) {
+	DisplayServerAndroid *ds = (DisplayServerAndroid *)DisplayServer::get_singleton();
+	if (ds) {
+		Vector<String> selected_paths;
+
+		jint length = env->GetArrayLength(p_selected_paths);
+		for (jint i = 0; i < length; ++i) {
+			jstring java_string = (jstring)env->GetObjectArrayElement(p_selected_paths, i);
+			const char *c_str = env->GetStringUTFChars(java_string, NULL);
+			selected_paths.push_back(String(c_str));
```

Comment:
```suggestion
			selected_paths.push_back(String::utf8(c_str));
```

Default `String` constructor is limited to ASCII.